### ISSUE TITLE
[Request] Logitech Onboard Memory Manager

### ISSUE THREAD

**Issue by Zliced13** [ScoopInstaller/Extras#9159]
### Criteria

For a package to be acceptable in the Extras bucket, it should be:

- [x] a well-known and widely used tool (e.g. if it's a GitHub project, it should have at least 100 stars and 50 forks)
- [x] the latest stable version of the program

### Information

**Name:** [name of new package]
logitech-omm

**Description:** [clear and concise details of what it is]
OnBoard Memory Manager (OMM) is a utility for pro gamers to quickly view, customize, and fine-tune the onboard memory of a compatible Logitech G mouse. 

**Homepage:** [a URL/link]
https://support.logi.com/hc/en-ca/articles/360059641133

**Download link(s):** [URL(s)/link(s)]
https://download01.logi.com/web/ftp/pub/techsupport/gaming/Onboard%20Memory%20Manager.exe

**Some indication of popularity/repute:** [GitHub stars/software reviews etc.]
https://www.reddit.com/r/LogitechG/comments/k4nvm4/new_logitech_onboard_memory_manager/
https://www.youtube.com/watch?v=a9WJFEnUOLc
https://nitter.dark.fail/onscreenlol/status/1432321731113410560?lang=en
https://nitter.42l.fr/kurtextrem/status/1334067955466964992

**Comment by brian6932** [comment]
Problem is, I don't believe there is a way to make a checkver for this, [I made a very simple manifest for it a while back](https://github.com/brian6932/dank-scoop/blob/master/bucket/onboard-memory-manager.json), but since logi's site completely requires js, and the installer is actually not accessible on the ftp server without a direct link, I'm at a loss.

I can PR this if you'd like @KrisSchneider however, it doesn't do a whole lot because of this unfortunately, if anyone knows some crazy way to fetch the checkver lmk.

**Comment by KrisSchneider** [comment]
Can it work without `"checkver"`? Is `"checkver"` really that important?

**Comment by brian6932** [comment]
I mean it will not auto update, you would have to manually update it @KrisSchneider
It very much works, however, it's not super useful imo 🤷
If someone can find a way to checkver OMM, I am super down to open a PR

**Comment by KrisSchneider** [comment]
Oh, so that is why. I don't mind waiting for someone to make a `"checkver"` for the manifest. Also, I am asking Logitech customer support if the app “Onboard Memory Manager” is still being developed?

**Comment by KrisSchneider** [comment]
A Logitech agent confirmed it will continue to be developed/supported by Logitech as long as there are people using it.
![image_1](https://user-images.githubusercontent.com/101912712/188494646-ec0a73ab-3109-4ac9-be43-b6e289d6ac1c.png)

**Comment by KrisSchneider** [comment]
Should I close this issue or should I keep it open?

**Comment by brian6932** [comment]
I'd keep it open, it's no resolved, maybe someone can figure out a way to checkver it or ask logitech to allow some type of way to do it, like openly hosting the files on their ftp server 🤷

**Comment by KrisSchneider** [comment]
I just got back with Logitech Support. I asked them if redistributing of software is allowed, and they said yes.
![(%pn)-W1r5t4_09-06-2022](https://user-images.githubusercontent.com/101912712/188772400-ebb95dbc-c937-4e42-bcbe-6036e06eab97.jpg)

**Comment by KrisSchneider** [comment]
I believe someone can upload a copy of the `.exe` file to this repository, https://github.com/ScoopInstaller/Binary. Would this solve the `"checkver"` problem? @brian6932

**Comment by brian6932** [comment]
No, that is a manual action, a checkver is used to check a version, and if the version is outdated, it will automatically update.
The problem is, the Logitech site is all js driven, there is no html you can regex, or json you can filter to auto update. They have an ftp server, however, the ftp server does not actually include the OMM files.

**Comment by brian6932** [comment]
At best you could use a 3rd party site like this [one](https://drivers.softpedia.com/get/KEYBOARD-and-MOUSE/LOGITECH/Logitech-Onboard-Memory-Manager-Utility-1-1-6805.shtml), to run a checkver, although it really isn't ideal.
It's better than what I have at least, so I'll use that for now

**Comment by KrisSchneider** [comment]
Sorry for using your time. I will just stick to manually downloading it from the website using my firefox browser.

**Comment by brian6932** [comment]
[Ok I managed to find a very hacky solution](https://github.com/brian6932/dank-scoop/blob/master/bucket/onboard-memory-manager.json)

**Comment by KrisSchneider** [comment]
I hope that works. Also thank you, @brian6932 for tackling this sticky issue.

**Comment by mpeter50** [comment]
I think there might be an easier (?) way for checking the version of the tool.

The support article is loaded from this JSON: https://logitech.zendesk.com/api/v2/help_center/de/articles/6449584392215.json
The article itself is in the articles[0].body value, in HTML form, in which the current version, last update date and other things can be found.
This way might be better, because you don't have to download the executable to check the version, you just need to obtain the JSON.

What do you think?

**Comment by Zliced13** [comment]
CC @brian6932

**Comment by mpeter50** [comment]
> EDIT: Also, how did you manage to find the link to this JSON file on Logitech's website?

I've opened a new tab in Firefox, opened the developer tools, and switched to the Network tab, so that it starts capturing network traffic.
Then I've loaded this page, where the tool can be downloaded: https://support.logi.com/hc/en-us/articles/6449584392215-Onboard-Memory-Manager
When loading finished, I did a search in the captured traffic for part of the version number seen on the page:

![image](https://user-images.githubusercontent.com/83356418/196287933-f095dcb8-2a57-482d-bdb5-b6162e0c6298.png)

Only part of it is important, because - even though this is not the case now - different components of the version number might be stored separately sometimes, for example in separate values of a JSON object.

Then I've gone through the results.
The first one is in the page HTML - weird, I don't think this result was here a few hours ago.
The second one is in the mentioned JSON. I've copied the response body to clipboard, inserted it into a JSOn formatter program I use, and searching again for the same part of the version number revealed where is the version number stored.

![image](https://user-images.githubusercontent.com/83356418/196288561-f81ab46b-826c-4c71-b580-53e366c58df1.png)
